### DISCUSSION TITLE
Configure title banner categories

### DISCUSSION THREAD

**Discussion by jcarpi** [quarto-dev/quarto-cli#3005] [Q&A]
Is there any way to hide certain title banner categories? For example, I don't want the full document abstract to show on every page of an HTML book!

**Comment by mcanouil** [comment]
Do you have an example?
Because I am almost certain you are actually not talking about the actual title banner, _i.e._, <https://quarto.org/docs/authoring/title-blocks.html#title-banners> which is the blue part with `title` and `description` fields of the picture below.
![](https://quarto.org/docs/authoring/images/document-title-block-banner.png)

**Reply by jcarpi** [reply]
Ah yes, I am talking about the "AUTHOR", "AFFILIATION", and "PUBLISHED" fields below. Is there any way to change which fields are shown?

**Reply by dragonstyle** [reply]
FWIW if the abstract is appearing on every page of a book, that seems like a bug! It should only appear on the main `index.qmd` page.

**Comment by mcanouil** [comment]
Yes, you'll have to redefine the `title-block.html` partial (see <https://quarto.org/docs/journals/templates.html#html-partials>).
For example, the default[^1] is the following:

```html
<header id="title-block-header">
<h1 class="title">$title$</h1>
$if(subtitle)$
<p class="subtitle">$subtitle$</p>

$endif$
$for(author)$
<p class="author">$author$</p>
$endfor$

$if(date)$
<p class="date">$date$</p>
$endif$
$if(abstract)$
<div class="abstract">
<div class="abstract-title">$abstract-title$</div>
$abstract$
</div>
$endif$
</header>
```

Then in your YAML header:

```yaml
format:
  html:
    template-partials:
      - title-block.html
```

[^1]: <https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/formats/html/pandoc/title-block.html>

**Reply by jcarpi** [reply]
Thank you! Unfortunately, when I apply this fix, the title-block-banner style goes away. My understanding is the HTML and the style (CSS, SCSS) are completely separate. Is this a bug?

**Reply by mcanouil** [reply]
Could you clarify what did not work? What did you try?
Without code, it's impossible to explain or fix anything, _i.e._, it might be a bug, or you made a syntax mistake, or anything.

**Comment by Gilquin** [comment]
Sorry to bump up this discussion, but I'm facing the same issue as @jcarpi.
Here is a full bug description.

## Bug description

The `title-block-banner` style becomes plain when redefining the `title-block.html` partial.

## Steps to reproduce

* Instantiate a new website project:
  ```bash
  quarto create-project test --type website
  ```
* Edit the file `test/about.qmd`:
  ```diff
  - ---
  - title: "About"
  - ---
  
  - About this site
  + ---
  + author: "John Doe"
  + categories:
  +   - title-block-banner
  +   - style
  + date: "2023-09-04"
  + description: "The title-block-banner style becomes plain when redefining the title-block.html partial."
  + title: "Title-block-banner issue"
  + title-block-banner: true
  + ---
  + 
  + Some lines.
  ```
* Change `theme: default` in `_quarto.yml` and preview the website

  `quarto preview test`:
  
  ![image](https://github.com/quarto-dev/quarto-cli/assets/12810238/05a6ac84-629b-4d23-9eb0-88f78f41b268)

* Next, edit the file `test/_quarto.yml` using the `title-block.html` file as suggested by @mcanouil :
  
  ```diff
  project:
  type: website
  
  website:
    title: "test"
    navbar:
      left:
        - href: index.qmd
          text: Home
        - about.qmd
  
  format:
    html:
  +   template-partials:
  +     - title-block.html
      theme: default
      css: styles.css
      toc: true
  ```
  The preview becomes:

  ![image](https://github.com/quarto-dev/quarto-cli/assets/12810238/500904d4-ba50-481e-a4b3-dd8af13d877f)

## Expected behavior

As indicated in the [documentation](https://quarto.org/docs/journals/templates.html#replacing-partials), one would expect the style of the `title-block-banner` to stay the same after replacing the partial with the exact same file.

## Actual behavior

* the style of the `title-block-banner` is removed,
* some elements disappear (description, categories)

## Your environment

OS: Ubuntu 22.04.3 LTS

## Quarto check output

```
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.1: OK
      Dart Sass version 1.55.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.3.433
      Path: /opt/quarto/bin

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.10.12
      Path: /usr/bin/python3
      Jupyter: (None)

      Jupyter is not available in this Python installation.
      Install with python3 -m pip install jupyter

      There is an unactivated Python environment in miniconda3. Did you forget to activate it?

[✓] Checking R installation...........OK
      Version: 4.3.1
      Path: /usr/lib/R
      LibPaths:
        - /home/lgilquin/R/x86_64-pc-linux-gnu-library/4.3
        - /usr/local/lib/R/site-library
        - /usr/lib/R/site-library
        - /usr/lib/R/library
      knitr: 1.41
      rmarkdown: 2.23

[✓] Checking Knitr engine render......OK
```

**Reply by dragonstyle** [reply]
I don't agree that it is reasonable to assume that replacing the title-block partial will still preserve the banner behavior. Often, the templates themselves contain HTML and accompanying CSS that control the appearance of the page, and may even use post processing to restructure or generate HTML. Without providing the appropriate classes and the resultant HTML structure, the appearance cannot be preserved. 

Our current model definitely assumes that if you replace the title-block HTML partial, you are taking responsibility for implementing its appearance. (In this particular case, the title block banner style requires additional coordination in post processing which isn't happening any longer since you are providing a custom template).

Can you say more about what you're actually trying to achieve - perhaps there is a way to get there without providing a custom template which also includes the banner treatment...

**Reply by Gilquin** [reply]
I'm not trying to be picky, but reading this part of the documentation:
> When Quarto renders a document with a partial, it will use the built in template but replace a portion of the template with the provided partial.

I would expect (wrongly) that replacing the `title-block.html` partial with the default one would yield the same result. I understand now that there is much more involved in the post-processing.

As for my use case, I just wanted to have a custom title-block-banner on a listing document without `author` and `date`. I cannot simply remove the `author` and `date` fields from the front matter because they are used as listing fields.

**Reply by cscheid** [reply]
> As for my use case, I just wanted to have a custom title-block-banner on a listing document without `author` and `date`.

As a hack, would adding a CSS rule that hides author and date work for you?

**Reply by Gilquin** [reply]
> As a hack, would adding a CSS rule that hides author and date work for you?

Yes, it would be great !

**Reply by dragonstyle** [reply]
Understandable - we do try to keep everything in the Pandoc templates as much as possible, but sometimes post processing simplifies things as in this case. It's something we'll try to improve over time.

One suggestion you could try would be to write a little LUA filter that just sets the author / date to metadata `nil` and use that filter only on pages which you'd like this data cleared for rendering. (The author data is normalized, so you actually would need to clear the values `by-affiliation` and `by-author` to prevent the template from seeing them).

I wonder if a better solution would be a way to control what fields are displayed in the title block metadata (e.g. you can provide a list of fields you'd like displayed). I was thinking about this for extending the metadata with custom values, but could also be useful in this context.

**Reply by cscheid** [reply]
> > As a hack, would adding a CSS rule that hides author and date work for you?
> 
> Yes, it would be great !

Sorry, I should have been clearer. If a CSS rule is enough, then that's something you can do by yourself, by identifying the CSS selectors that pick the author content in a title block (using your browser's dev tools) and adding a `display: none` style to them in a CSS class of your own.

**Reply by Gilquin** [reply]
> I wonder if a better solution would be a way to control what fields are displayed in the title block metadata (e.g. you can provide a list of fields you'd like displayed). I was thinking about this for extending the metadata with custom values, but could also be useful in this context.

That would be really useful.
I will dig into the LUA filter solution, thanks !

**Reply by Gilquin** [reply]
@cscheid 
>Sorry, I should have been clearer. If a CSS rule is enough, then that's something you can do by yourself, by identifying the CSS selectors that pick the author content in a title block (using your browser's dev tools) and adding a display: none style to them in a CSS class of your own.

No, it's my bad, sorry ! 
Thanks for the pointer !

**Reply by Gilquin** [reply]
Here is a simple working Lua filter to remove author and date as per @dragonstyle suggested:
```lua
-- Purpose: Clean author and date from the front matter
function Meta(m)
  if m.date ~= nil then
    m.date = nil
  end
  if m.author ~= nil then
    m.author = nil
    m["by-author"] = nil
    m["by-affiliation"] = nil
  end
  return m
end
```
Thanks again for the help!

**Reply by the-solipsist** [reply]
> I wonder if a better solution would be a way to control what fields are displayed in the title block metadata (e.g. you can provide a list of fields you'd like displayed). I was thinking about this for extending the metadata with custom values, but could also be useful in this context

I think having the ability to have custom metadata read by quarto would be very helpful for both title blocks and for listings.

What would be amazing is if I could get Quarto to read a CSL YAML reference inside a document and use that to fill in title, date, author, in addition to generating the BibTeX citation as it currently does. (Currently, I'm duplicating many of the fields in my YAML metadata, at the base level and within 'citation:'.)
